NodeClassifier/utils/featurizer.py

```diff
@@ -56,9 +55,15 @@ def extract_features(session_dict, capture_source=None, max_port=None):
     # Iterate over all sessions and aggregate the info
     other_ips = defaultdict(int)
     for key, session in session_dict.items():
-        address_1, port_1 = key[0].split(':')
-        address_2, port_2 = key[1].split(':')
-
+        splitter_index = key[0].rindex(':')
+        address_1 = key[0][0:splitter_index]
+        port_1 = key[0][splitter_index + 1:]
+
+        splitter_index = key[1].rindex(':')
+        address_2 = key[1][0:splitter_index]
+        port_2 = key[1][splitter_index + 1:]
+        
```

Comment:
Out of curiosity.. how come you changed this from the original?

Reply:
Yeah, that's beceause we have now IPv6 which include `:` in its IP part. 
so if the key is ipv6:port, the split will return a list with length > 2. 

I think there is a better way to do that like 

```
key_str = key[0].split(':')
address_1 = ':'.join(key_str[0:-1])
port_1 = key_str[-1]

key_str = key[1].split(':')
address_2 = ':'.join(key_str[0:-1])
port_2 = key_str[-1]

```
what do you think?

Reply:
You know... good point. 🤔

Reply:
Thanks so much for your contribution; this is a really great add to the project. I think I have an idea that'll maintain readability in both the code and as output, so I'll work on that as soon as I merge your changes.

Thanks again!